src/main/python/camdkit/framework.py

```diff
@@ -16,6 +16,7 @@
 class Sampling(Enum):
   STATIC = "Static"
   REGULAR = "Regular"
+  DYNAMIC = "Dynamic"
```

Comment:
What is the difference between "Dynamic" and "Regular"?

Reply:
Regular seemed to be a case that was used for arrays so I created a new case so as not to break any of that...

Reply:
What is dynamic then?

Reply:
Ah I did not understand regularly sampled. I took regular to mean 'normal' - oops! I added the DYNAMIC case so as not to touch the existing code too much. (I added recursion on ParameterContainer::to_json e.g.). Perhaps the framework needs to be refactored to handle objects and arrays in a more generic way? I'm happy to have a go at this, but is it a separate (later) PR?

Reply:
> Perhaps the framework needs to be refactored to handle objects and arrays in a more generic way?

A [parameter](https://github.com/SMPTE/ris-osvp-metadata-camdkit/blob/0e56559d7d53e0c9a585addc676c64afaca1e065/src/main/python/camdkit/framework.py#L27) can be anything, including an array. What do you mean by "more generic"?

>  I'm happy to have a go at this, but is it a separate (later) PR?

Yes.

Reply:
> > Perhaps the framework needs to be refactored to handle objects and arrays in a more generic way?
> 
> A [parameter](https://github.com/SMPTE/ris-osvp-metadata-camdkit/blob/0e56559d7d53e0c9a585addc676c64afaca1e065/src/main/python/camdkit/framework.py#L27) can be anything, including an array. What do you mean by "more generic"?

Not currently - if I just set my parameters to regular then lines like this won't work:
```
        elif desc.sampling is Sampling.REGULAR:
          self._values[prop] = tuple(map(desc.from_json, json_value))
```

Reply:
Each parameter in the model must be defined in https://github.com/SMPTE/ris-osvp-metadata-camdkit/blob/main/src/main/python/camdkit/model.py

Reply:
See usage at https://github.com/SMPTE/ris-osvp-metadata-camdkit/blob/main/src/test/python/test_model.py

Reply:
Yes I have created my parameters in trackerkit/model.py - I will look at the tests to see what I was doing wrong. Thanks.

Reply:
Ok I get it now. A Clip has tuples of 'regular' data. Whereas my 'Frame' has a single data point, but it is not 'static' as such. So I can now answer your original question. My concept of 'Dynamic' data is one that is changing every one of my 'Frames' I will comment the enum accordingly. We may want to rethink this in the future, but are you happy to proceed with the 'Dynamic' option for now?

Reply:
Just to close off this thread - after much discussion on Slack I will be reworking the architecture of this to integrate better with the existing model, rather than work alongside it.